ste/getProxy_windows.go

```diff
@@ -0,0 +1,35 @@
+package ste
+
+import (
+	"golang.org/x/net/http/httpproxy"
+	"golang.org/x/sys/windows/registry"
+	"net/http"
+	"net/url"
+)
+
+func getProxy() func(*http.Request) (*url.URL, error) {
+	if prox := httpproxy.FromEnvironment(); (prox.HTTPProxy != "" || prox.HTTPSProxy != "") && prox.NoProxy == "" {
+		return http.ProxyFromEnvironment
+		//Option to default to env vars, with a fallback to env vars if grabbing from IE/Edge fails
+	}
+
+	key, err := registry.OpenKey(registry.LOCAL_MACHINE, `Software\Microsoft\Windows\CurrentVersion\Internet Settings`, registry.QUERY_VALUE)
+	if err != nil {
+		return http.ProxyFromEnvironment
+	}
+	defer key.Close()
+
+	strURL, _, err := key.GetStringValue("ProxyServer")
```

Comment:
When I was looking at the IE proxy settings window on my machine just now, I realized we are not covering all the cases.  
1. The most minor (in terms of consequence) is that we are not covering the exclusion list that can be storage in ProxyOverride registry value.
2. More importantly, we are not respecting the ProxyEnable flag. That's important, because its possible for ProxyServer to be populated, but ProxyEnable set to false.
3. But potentially the biggest issue of all is that we can't easily support the auto-configuration option. (which I suspect is actually probably more commonly used that the ProxyServer value).

So.... to address the above, could you have a look at https://github.com/mattn/go-ieproxy and (a) see if it looks like a good way to address 2 and 3 above (and maybe 1). and (b) do a quick web search to see if that particular project looks like the best one to use, or if there are other strong contenders.

cc @zezha-msft in case he has any suggestions.